### PR TITLE
OCPBUGS-44968: Reconcile SecretProvider for CSO on ARO HCP

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Reconcile the SecretProviderClass for the cluster storage operator (CSO) for ARO HCP deployments. The SecretProviderClass is used by the Secrets Store CSI driver to mount a certificate to a volume in the azure-disk-csi-controller and azure-file-csi-controller pod deployments.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes [OCPBUGS-44968](https://issues.redhat.com/browse/OCPBUGS-44968)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by bryan-cox** [comment]
Needs the following PRs before we can continue with this one:
- https://github.com/openshift/hypershift/pull/4877
- https://github.com/openshift/cluster-storage-operator/pull/517
- https://github.com/openshift/csi-operator/pull/290

File: control-plane-operator/controllers/hostedcontrolplane/storage/operator.go [hunk 1]
```diff
@@ -31,6 +32,21 @@ func ReconcileOperatorDeployment(
...
 			deployment.Spec.Template.Spec.Containers[i].Image = params.StorageOperatorImage
 			params.ImageReplacer.replaceEnvVars(deployment.Spec.Template.Spec.Containers[i].Env)
+
+			if azureutil.IsAroHCP() {
+				if deployment.Spec.Template.Spec.Containers[i].Env == nil {
+					deployment.Spec.Template.Spec.Containers[i].Env = make([]corev1.EnvVar, 0)
+				}
+				deployment.Spec.Template.Spec.Containers[i].Env = append(deployment.Spec.Template.Spec.Containers[i].Env,
+					corev1.EnvVar{
+						Name:  "ARO_HCP_SECRET_PROVIDER_CLASS_FOR_DISK",
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/storage/operator.go:42]: can we // doc and link who is consuming this magic variable?

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 2]
```diff
@@ -4986,6 +4986,34 @@ func (r *HostedControlPlaneReconciler) reconcileClusterStorageOperator(ctx conte
 		}); err != nil {
 			return fmt.Errorf("failed to reconcile Azure File Secret Provider Class: %w", err)
 		}
+
+		// Get the credentials secret so we can retrieve the tenant ID for the configuration
+		credentialsSecret := manifests.AzureCredentialInformation(hcp.Namespace)
+		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(credentialsSecret), credentialsSecret); err != nil {
+			return fmt.Errorf("failed to get Azure credentials secret: %w", err)
+		}
+		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(credentialsSecret), credentialsSecret); err != nil {
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:4995]: remove duplicate code

**Review by csrwng — COMMENTED** [review]
Just one comment, otherwise lgtm

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go (hunk 2 above)
Comment (bryan-cox) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go]: Whoops! Not sure how that happened. It's been removed now.

**Pull request merged**